### ISSUE TITLE
Android TV unable to connect

### ISSUE THREAD

**Issue by niklastanner** [jellyfin/jellyfin-androidtv#3486]
### Describe the bug

I am unable to connect to my jellyfin server from my Android TV app.

# Setup
I've got Jellyfin running inside a kubernetes Cluster.  
I use traefik as reverse proxy. exposed ports are 80, 443, 22. I'm using duckdns.org as dyndns. (let's assume example.duckdns.org for now, as I won't expose the real domain here).
Jellyfin is listening to media.example.duckdns.org.

traefik makes automated ACME http01 challenges for media.example.duckdns.org. The certificate is valid.
The deployment is using the image `docker.io/jellyfin/jellyfin:2024031105`

The deployment, service and ingress expose port 8096 which is mapped to 80 (for HTTP) and 443 (for HTTPS) by traefik.
HTTPS on jellyfin itself is disabled (Port 8920 disabled), as the entire traffic is encrypted by the reverse proxy itself using the Let's Encrypt certificate.

Local network connection is disabled entirely (for now). All devices have to connect to the jellyfin server using `(http or https)://media.example.duckdns.org`

Jellyfin has the following environment variables set within the deployment:
```
env:
  - name: JELLYFIN_PublishedServerUrl
    value: media.example.duckdns.org
  - name: JELLYFIN_LOG_DIR
    value: /cache/log
```

# Issue
First of all, this is a new installation. My problem existed from the beginning and it never worked on the TV.

I can connect to the server and stream content using the following clients:
- Windows Application
- Browser on Windows (connecting to https://media.example.duckdns.org)
- Android App

I cannot connect to the server using the following client:
- Android TV on my Sony television (app version `0.16.7`)

## What I do to provoke the issue
1. Open the Jellyfin app on my TV
2. Connect to server media.example.duckdns.org (http:// is not possible, it always gets overridden by the app to https://)
3. Create user
4. Select Authentication using Quick Connect
5. Open Jellyfin on my android phone and enter the Quick Connect ID

After entering the Quick Connect ID and confirming, the TV shows a loading animation and instantly displays `Unable to connect to server`. This is where I'm stuck right now.

## A wild guess
Maybe the android tv is trying to connect to the server on a port that is not allowed by the firewall/router (any port except 80, 443, 22) and therefore can't reach the server. But I couldn't find any documentation backing this up and I also didn't see any traffic on the router regarding ports 8096 or 8920, only 443. But at this point I'm not sure what to check exactly.

### Logs

The relevant jellyfin server logs after authenticating using Quick Connect:
```
[INF] [24] Emby.Server.Implementations.Session.SessionManager: Current/Max sessions for user test: 2/0
[INF] [24] Emby.Server.Implementations.Session.SessionManager: Creating new access token for user <UUID of user test>
[DBG] [19] Emby.Server.Implementations.QuickConnect.QuickConnectManager: Authorizing device with code 867591 to login as user <UUID of user test>
[DBG] [20] Emby.Server.Implementations.Session.SessionWebSocketListener: Watching 1 WebSockets.
```

There is no other log message except for the WebSocket timeout a few minutes later.

Traefik logs (tried to isolate the TV portion as much as possible but may contain logs from other clients too):
```
{ "ClientAddr": "10.1.21.1:42909", "ClientHost": "10.1.21.1", "ClientPort": "42909", "ClientUsername": "-", "DownstreamContentSize": 7473, "DownstreamStatus": 200, "Duration": 4331584, "OriginContentSize": 7473, "OriginDuration": 4268399, "OriginStatus": 200, "Overhead": 63185, "RequestAddr": "media.example.duckdns.org", "RequestContentSize": 0, "RequestCount": 797, "RequestHost": "media.example.duckdns.org", "RequestMethod": "GET", "RequestPath": "/ScheduledTasks?IsEnabled=true", "RequestPort": "-", "RequestProtocol": "HTTP/2.0", "RequestScheme": "https", "RetryAttempts": 0, "RouterName": "websecure-jellyfin-jellyfin-media-example-duckdns-org@kubernetes", "ServiceAddr": "10.1.21.96:8096", "ServiceName": "jellyfin-jellyfin-default@kubernetes", "ServiceURL": { "Scheme": "http", "Opaque": "", "User": null, "Host": "10.1.21.96:8096", "Path": "", "RawPath": "", "OmitHost": false, "ForceQuery": false, "RawQuery": "", "Fragment": "", "RawFragment": "" }, "StartLocal": "2024-04-05T12:54:40.574544327Z", "StartUTC": "2024-04-05T12:54:40.574544327Z", "TLSCipher": "TLS_AES_128_GCM_SHA256", "TLSVersion": "1.3", "entryPointName": "websecure", "level": "info", "msg": "", "time": "2024-04-05T12:54:40Z"}
{ "ClientAddr": "10.1.21.1:42585", "ClientHost": "10.1.21.1", "ClientPort": "42585", "ClientUsername": "-", "DownstreamContentSize": 2, "DownstreamStatus": 200, "Duration": 4322658, "OriginContentSize": 2, "OriginDuration": 4247714, "OriginStatus": 200, "Overhead": 74944, "RequestAddr": "media.example.duckdns.org", "RequestContentSize": 0, "RequestCount": 798, "RequestHost": "media.example.duckdns.org", "RequestMethod": "GET", "RequestPath": "/Users/Public", "RequestPort": "-", "RequestProtocol": "HTTP/2.0", "RequestScheme": "https", "RetryAttempts": 0, "RouterName": "websecure-jellyfin-jellyfin-media-example-duckdns-org@kubernetes", "ServiceAddr": "10.1.21.96:8096", "ServiceName": "jellyfin-jellyfin-default@kubernetes", "ServiceURL": { "Scheme": "http", "Opaque": "", "User": null, "Host": "10.1.21.96:8096", "Path": "", "RawPath": "", "OmitHost": false, "ForceQuery": false, "RawQuery": "", "Fragment": "", "RawFragment": "" }, "StartLocal": "2024-04-05T12:54:41.297997559Z", "StartUTC": "2024-04-05T12:54:41.297997559Z", "TLSCipher": "TLS_AES_128_GCM_SHA256", "TLSVersion": "1.3", "entryPointName": "websecure", "level": "info", "msg": "", "time": "2024-04-05T12:54:41Z"}
{ "ClientAddr": "10.1.21.1:42585", "ClientHost": "10.1.21.1", "ClientPort": "42585", "ClientUsername": "-", "DownstreamContentSize": 2, "DownstreamStatus": 200, "Duration": 3761421, "OriginContentSize": 2, "OriginDuration": 3689255, "OriginStatus": 200, "Overhead": 72166, "RequestAddr": "media.example.duckdns.org", "RequestContentSize": 0, "RequestCount": 799, "RequestHost": "media.example.duckdns.org", "RequestMethod": "GET", "RequestPath": "/Users/Public", "RequestPort": "-", "RequestProtocol": "HTTP/2.0", "RequestScheme": "https", "RetryAttempts": 0, "RouterName": "websecure-jellyfin-jellyfin-media-example-duckdns-org@kubernetes", "ServiceAddr": "10.1.21.96:8096", "ServiceName": "jellyfin-jellyfin-default@kubernetes", "ServiceURL": { "Scheme": "http", "Opaque": "", "User": null, "Host": "10.1.21.96:8096", "Path": "", "RawPath": "", "OmitHost": false, "ForceQuery": false, "RawQuery": "", "Fragment": "", "RawFragment": "" }, "StartLocal": "2024-04-05T12:54:41.336638056Z", "StartUTC": "2024-04-05T12:54:41.336638056Z", "TLSCipher": "TLS_AES_128_GCM_SHA256", "TLSVersion": "1.3", "entryPointName": "websecure", "level": "info", "msg": "", "time": "2024-04-05T12:54:41Z"}
{ "ClientAddr": "10.1.21.1:42585", "ClientHost": "10.1.21.1", "ClientPort": "42585", "ClientUsername": "-", "DownstreamContentSize": 65, "DownstreamStatus": 200, "Duration": 1103308, "OriginContentSize": 65, "OriginDuration": 1032845, "OriginStatus": 200, "Overhead": 70463, "RequestAddr": "media.example.duckdns.org", "RequestContentSize": 0, "RequestCount": 800, "RequestHost": "media.example.duckdns.org", "RequestMethod": "GET", "RequestPath": "/Branding/Configuration", "RequestPort": "-", "RequestProtocol": "HTTP/2.0", "RequestScheme": "https", "RetryAttempts": 0, "RouterName": "websecure-jellyfin-jellyfin-media-example-duckdns-org@kubernetes", "ServiceAddr": "10.1.21.96:8096", "ServiceName": "jellyfin-jellyfin-default@kubernetes", "ServiceURL": { "Scheme": "http", "Opaque": "", "User": null, "Host": "10.1.21.96:8096", "Path": "", "RawPath": "", "OmitHost": false, "ForceQuery": false, "RawQuery": "", "Fragment": "", "RawFragment": "" }, "StartLocal": "2024-04-05T12:54:41.343893233Z", "StartUTC": "2024-04-05T12:54:41.343893233Z", "TLSCipher": "TLS_AES_128_GCM_SHA256", "TLSVersion": "1.3", "entryPointName": "websecure", "level": "info", "msg": "", "time": "2024-04-05T12:54:41Z"}
{ "ClientAddr": "10.1.21.1:42585", "ClientHost": "10.1.21.1", "ClientPort": "42585", "ClientUsername": "-", "DownstreamContentSize": 214, "DownstreamStatus": 200, "Duration": 1055715, "OriginContentSize": 214, "OriginDuration": 993050, "OriginStatus": 200, "Overhead": 62665, "RequestAddr": "media.example.duckdns.org", "RequestContentSize": 0, "RequestCount": 801, "RequestHost": "media.example.duckdns.org", "RequestMethod": "GET", "RequestPath": "/System/Info/Public", "RequestPort": "-", "RequestProtocol": "HTTP/2.0", "RequestScheme": "https", "RetryAttempts": 0, "RouterName": "websecure-jellyfin-jellyfin-media-example-duckdns-org@kubernetes", "ServiceAddr": "10.1.21.96:8096", "ServiceName": "jellyfin-jellyfin-default@kubernetes", "ServiceURL": { "Scheme": "http", "Opaque": "", "User": null, "Host": "10.1.21.96:8096", "Path": "", "RawPath": "", "OmitHost": false, "ForceQuery": false, "RawQuery": "", "Fragment": "", "RawFragment": "" }, "StartLocal": "2024-04-05T12:54:41.353244878Z", "StartUTC": "2024-04-05T12:54:41.353244878Z", "TLSCipher": "TLS_AES_128_GCM_SHA256", "TLSVersion": "1.3", "entryPointName": "websecure", "level": "info", "msg": "", "time": "2024-04-05T12:54:41Z"}
{ "ClientAddr": "10.1.21.1:42585", "ClientHost": "10.1.21.1", "ClientPort": "42585", "ClientUsername": "-", "DownstreamContentSize": 283, "DownstreamStatus": 200, "Duration": 2523854, "OriginContentSize": 283, "OriginDuration": 2450743, "OriginStatus": 200, "Overhead": 73111, "RequestAddr": "media.example.duckdns.org", "RequestContentSize": 0, "RequestCount": 802, "RequestHost": "media.example.duckdns.org", "RequestMethod": "GET", "RequestPath": "/QuickConnect/Initiate", "RequestPort": "-", "RequestProtocol": "HTTP/2.0", "RequestScheme": "https", "RetryAttempts": 0, "RouterName": "websecure-jellyfin-jellyfin-media-example-duckdns-org@kubernetes", "ServiceAddr": "10.1.21.96:8096", "ServiceName": "jellyfin-jellyfin-default@kubernetes", "ServiceURL": { "Scheme": "http", "Opaque": "", "User": null, "Host": "10.1.21.96:8096", "Path": "", "RawPath": "", "OmitHost": false, "ForceQuery": false, "RawQuery": "", "Fragment": "", "RawFragment": "" }, "StartLocal": "2024-04-05T12:54:43.299618006Z", "StartUTC": "2024-04-05T12:54:43.299618006Z", "TLSCipher": "TLS_AES_128_GCM_SHA256", "TLSVersion": "1.3", "entryPointName": "websecure", "level": "info", "msg": "", "time": "2024-04-05T12:54:43Z"}
{ "ClientAddr": "10.1.21.1:42585", "ClientHost": "10.1.21.1", "ClientPort": "42585", "ClientUsername": "-", "DownstreamContentSize": 283, "DownstreamStatus": 200, "Duration": 1292121, "OriginContentSize": 283, "OriginDuration": 1228418, "OriginStatus": 200, "Overhead": 63703, "RequestAddr": "media.example.duckdns.org", "RequestContentSize": 0, "RequestCount": 805, "RequestHost": "media.example.duckdns.org", "RequestMethod": "GET", "RequestPath": "/QuickConnect/Connect?secret=<REDACTED>", "RequestPort": "-", "RequestProtocol": "HTTP/2.0", "RequestScheme": "https", "RetryAttempts": 0, "RouterName": "websecure-jellyfin-jellyfin-media-example-duckdns-org@kubernetes", "ServiceAddr": "10.1.21.96:8096", "ServiceName": "jellyfin-jellyfin-default@kubernetes", "ServiceURL": { "Scheme": "http", "Opaque": "", "User": null, "Host": "10.1.21.96:8096", "Path": "", "RawPath": "", "OmitHost": false, "ForceQuery": false, "RawQuery": "", "Fragment": "", "RawFragment": "" }, "StartLocal": "2024-04-05T12:54:48.314666834Z", "StartUTC": "2024-04-05T12:54:48.314666834Z", "TLSCipher": "TLS_AES_128_GCM_SHA256", "TLSVersion": "1.3", "entryPointName": "websecure", "level": "info", "msg": "", "time": "2024-04-05T12:54:48Z"}
{ "ClientAddr": "10.1.21.1:26364", "ClientHost": "10.1.21.1", "ClientPort": "26364", "ClientUsername": "-", "DownstreamContentSize": 4, "DownstreamStatus": 200, "Duration": 201974578, "OriginContentSize": 4, "OriginDuration": 201908523, "OriginStatus": 200, "Overhead": 66055, "RequestAddr": "media.example.duckdns.org", "RequestContentSize": 0, "RequestCount": 806, "RequestHost": "media.example.duckdns.org", "RequestMethod": "POST", "RequestPath": "/QuickConnect/Authorize?code=196413\u0026userId=<UUID of test without hyphen>", "RequestPort": "-", "RequestProtocol": "HTTP/2.0", "RequestScheme": "https", "RetryAttempts": 0, "RouterName": "websecure-jellyfin-jellyfin-media-example-duckdns-org@kubernetes", "ServiceAddr": "10.1.21.96:8096", "ServiceName": "jellyfin-jellyfin-default@kubernetes", "ServiceURL": { "Scheme": "http", "Opaque": "", "User": null, "Host": "10.1.21.96:8096", "Path": "", "RawPath": "", "OmitHost": false, "ForceQuery": false, "RawQuery": "", "Fragment": "", "RawFragment": "" }, "StartLocal": "2024-04-05T12:54:49.41947031Z", "StartUTC": "2024-04-05T12:54:49.41947031Z", "TLSCipher": "TLS_AES_128_GCM_SHA256", "TLSVersion": "1.3", "entryPointName": "websecure", "level": "info", "msg": "", "time": "2024-04-05T12:54:49Z"}
{ "ClientAddr": "10.1.21.1:42585", "ClientHost": "10.1.21.1", "ClientPort": "42585", "ClientUsername": "-", "DownstreamContentSize": 282, "DownstreamStatus": 200, "Duration": 1031198, "OriginContentSize": 282, "OriginDuration": 965827, "OriginStatus": 200, "Overhead": 65371, "RequestAddr": "media.example.duckdns.org", "RequestContentSize": 0, "RequestCount": 807, "RequestHost": "media.example.duckdns.org", "RequestMethod": "GET", "RequestPath": "/QuickConnect/Connect?secret=<REDACTED>", "RequestPort": "-", "RequestProtocol": "HTTP/2.0", "RequestScheme": "https", "RetryAttempts": 0, "RouterName": "websecure-jellyfin-jellyfin-media-example-duckdns-org@kubernetes", "ServiceAddr": "10.1.21.96:8096", "ServiceName": "jellyfin-jellyfin-default@kubernetes", "ServiceURL": { "Scheme": "http", "Opaque": "", "User": null, "Host": "10.1.21.96:8096", "Path": "", "RawPath": "", "OmitHost": false, "ForceQuery": false, "RawQuery": "", "Fragment": "", "RawFragment": "" }, "StartLocal": "2024-04-05T12:54:53.330087091Z", "StartUTC": "2024-04-05T12:54:53.330087091Z", "TLSCipher": "TLS_AES_128_GCM_SHA256", "TLSVersion": "1.3", "entryPointName": "websecure", "level": "info", "msg": "", "time": "2024-04-05T12:54:53Z"}
{ "ClientAddr": "10.1.21.1:42585", "ClientHost": "10.1.21.1", "ClientPort": "42585", "ClientUsername": "-", "DownstreamContentSize": 3127, "DownstreamStatus": 200, "Duration": 1723285, "OriginContentSize": 3127, "OriginDuration": 1659805, "OriginStatus": 200, "Overhead": 63480, "RequestAddr": "media.example.duckdns.org", "RequestContentSize": 77, "RequestCount": 808, "RequestHost": "media.example.duckdns.org", "RequestMethod": "POST", "RequestPath": "/Users/AuthenticateWithQuickConnect", "RequestPort": "-", "RequestProtocol": "HTTP/2.0", "RequestScheme": "https", "RetryAttempts": 0, "RouterName": "websecure-jellyfin-jellyfin-media-example-duckdns-org@kubernetes", "ServiceAddr": "10.1.21.96:8096", "ServiceName": "jellyfin-jellyfin-default@kubernetes", "ServiceURL": { "Scheme": "http", "Opaque": "", "User": null, "Host": "10.1.21.96:8096", "Path": "", "RawPath": "", "OmitHost": false, "ForceQuery": false, "RawQuery": "", "Fragment": "", "RawFragment": "" }, "StartLocal": "2024-04-05T12:54:53.34177674Z", "StartUTC": "2024-04-05T12:54:53.34177674Z", "TLSCipher": "TLS_AES_128_GCM_SHA256", "TLSVersion": "1.3", "entryPointName": "websecure", "level": "info", "msg": "", "time": "2024-04-05T12:54:53Z"}
{ "ClientAddr": "10.1.21.1:54663", "ClientHost": "10.1.21.1", "ClientPort": "54663", "ClientUsername": "-", "DownstreamContentSize": 0, "DownstreamStatus": 0, "Duration": 57345549430, "OriginContentSize": 0, "OriginDuration": 57345463690, "OriginStatus": 0, "Overhead": 85740, "RequestAddr": "media.example.duckdns.org", "RequestContentSize": 0, "RequestCount": 771, "RequestHost": "media.example.duckdns.org", "RequestMethod": "GET", "RequestPath": "/socket?api_key=<REDACTED>\u0026deviceId=a0abcf4511b3b1bf", "RequestPort": "-", "RequestProtocol": "HTTP/1.1", "RequestScheme": "https", "RetryAttempts": 0, "RouterName": "websecure-jellyfin-jellyfin-media-example-duckdns-org@kubernetes", "ServiceAddr": "10.1.21.96:8096", "ServiceName": "jellyfin-jellyfin-default@kubernetes", "ServiceURL": { "Scheme": "http", "Opaque": "", "User": null, "Host": "10.1.21.96:8096", "Path": "", "RawPath": "", "OmitHost": false, "ForceQuery": false, "RawQuery": "", "Fragment": "", "RawFragment": "" }, "StartLocal": "2024-04-05T12:54:04.625265938Z", "StartUTC": "2024-04-05T12:54:04.625265938Z", "TLSCipher": "TLS_AES_128_GCM_SHA256", "TLSVersion": "1.3", "entryPointName": "websecure", "level": "info", "msg": "", "time": "2024-04-05T12:55:01Z"}
```

### Application version

0.16.7

### Where did you install the app from?

Google Play

### Device information

Sony BRAVIA 4K VH2

### Android version

Android 10

### Jellyfin server version

10.9.0

**Comment by nielsvanvelzen** [comment]
Which specific version of 10.9 are you using? You need at least 20240405 for login to work.

See #3467

**Comment by niklastanner** [comment]
Hi @nielsvanvelzen thanks for the quick response.
I was using `docker.io/jellyfin/jellyfin:2024031105` if that's what you're looking for.
I did not find a tag `20240405` so I'm not sure if you're asking for something else.

But just to go sure, I updated to `docker.io/jellyfin/jellyfin:2024040105` published 4 days ago.
This solved the problem, I can now connect on my Android TV too, thanks a lot!
